python/spacewalk/server/dbConfig.py

```diff
@@ -0,0 +1,64 @@
+#  pylint: disable=missing-module-docstring,invalid-name
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025 SUSE LLC
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+from . import rhnSQL
+
+
+def value(key):
+    """
+    Return the value of the given key. If the value is not defined, return the default value
+    When the key is not found, an AttributeError is raised
+    """
+
+    def sanitize_value(key, val):
+        """
+        attempt to convert a string value to the proper type
+        """
+        convert_table = {
+            "PSW_CHECK_SPECIAL_CHARACTERS": str,
+        }
+        val = val.strip()
+
+        if convert_table.get(key):
+            try:
+                val = convert_table.get(key)(val)
+            except ValueError:
+                pass
+        else:
```

Comment:
So we want:

- For specific values, we want specific type defined in the inner dict
- Otherwise, we want an int,
- Unless it's a float.
- But default return from a DB is string (right?), so if it is neither an int or a float, it'll be a string anyways
- Finally, if it is an empty string, return None

Is that accurate? (I'm asking because this inner method seems confusing to me, and like we should be able to make it simpler than if/else-with-try-except-try-except)

Reply:
yes, we try to convert to the real type int, float or string.
There may special keys which always must be converted to a specified type.
Example: there is a password and you decide to use "1234" as password. We want it to be a string, but that mechanism would convert it to a int if we would not have the convert_table.